src/org/labkey/test/tests/SampleSetTest.java

```diff
@@ -617,6 +633,85 @@ public void deleteLineageParent() throws IOException, CommandException
         dgen.deleteDomain(createDefaultConnection(true));
     }
 
+
+    @Test
+    public void testStringLookupFields() throws IOException, CommandException
+    {
+        String sampleSetName = "10000Samples";
+        String listName = "MainList";
+
+        goToProjectHome();
+        new PortalHelper(this).addWebPart("Lists");
+
+        log("Creating the sample set of 10000 samples");
+        FieldDefinition.LookupInfo lookupInfo = new FieldDefinition.LookupInfo(getProjectName(), "exp.materials", sampleSetName);
+        TestDataGenerator dgen = new TestDataGenerator(lookupInfo)
+                .withColumns(List.of(
+                        TestDataGenerator.simpleFieldDef("name", FieldDefinition.ColumnType.String),
+                        TestDataGenerator.simpleFieldDef("label", FieldDefinition.ColumnType.String)));
+        dgen.addDataSupplier("label", () -> dgen.randomString(10))
+                .withGeneratedRows(10000);
+        dgen.createDomain(createDefaultConnection(true), "SampleSet");
+        dgen.insertRows(createDefaultConnection(true), dgen.getRows());
+
+        log("Waiting for the sample data to get generated");
+        goToProjectHome();
+        waitAndClickAndWait(Locator.linkWithText(sampleSetName));
+        waitForTextWithRefresh(400000, "1 - 100 of 10,000");
```

Comment:
Why so many rows? That seems like a quantity you'd use for perf or stress testing.

Reply:
Perhaps I failed to submit this comment earlier, but this waiting does not seem to be necessary, even though this "feature" requires 10000 rows in order to be activated. The insertRows call is a blocking call and it looks like it  returns a structure that can be used to verify that the expected number of rows have been inserted.  I hope the rest of the test doesn't require that this UI has been rendered.

Reply:
@labkey-tchad Added the commenting explaining the reason to add 10000 samples

@labkey-susanh Changing the wait condition